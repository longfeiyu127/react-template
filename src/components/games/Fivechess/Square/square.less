
.g-fic-square {
  background: transparent;
  box-sizing: border-box;
  // font-size: 24px;
  font-weight: bold;
  // line-height: 100%;
  height: 100%;
  padding: 0;
  text-align: center;
  width: 100%;
  .squareItem{
    height: 100%;
    .squarePiece {
      height: 100%;
    }
    .squareBlack{
      background: url('../../../../assets/images/games/fivechess/black.png') no-repeat center;
      background-size: 90% 90%;
    }
    .squareWhite{
      background: url('../../../../assets/images/games/fivechess/white.png') no-repeat center;
      background-size: 90% 90%;
    }
  }
  .squareWin{
    border-radius: 50%;
    background: radial-gradient(rgba(5, 246, 253, 0) 0%, rgba(5, 246, 253, 1) 90%,  rgba(255, 255, 255, 0) 100%);
  }
}

// .g-tic-square:focus {
//   outline: none;
// }