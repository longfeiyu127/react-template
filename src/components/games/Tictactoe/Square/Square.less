
.g-tic-square {
  background: transparent;
  box-sizing: border-box;
  font-size: 24px;
  font-weight: bold;
  line-height: 100%;
  height: 100%;
  padding: 0;
  text-align: center;
  width: 100%;
  .squareItem{
    height: 100%;
  }
  .squareItemX{
    background: url('../../../../assets/images/games/tictactoe/x.png') no-repeat center;
    background-size: 70% 70%;
  }
  .squareItemO{
    background: url('../../../../assets/images/games/tictactoe/o.png') no-repeat center;
    background-size: 70% 70%;
  }
}

// .g-tic-square:focus {
//   outline: none;
// }