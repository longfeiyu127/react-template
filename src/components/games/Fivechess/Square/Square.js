import React from 'react';
import './square.less';

export default function Square(props) {
  // console.log(props)
  // eslint-disable-next-line
  const { onClick, inWinLine, value } = props;
  return (
    <div className="g-fic-square" onClick={onClick}>
      {/* <div className="squareItem squareBlack"></div> */}
      <div className={['squareItem', inWinLine ? 'squareWin' : ''].join(' ')}>
        <div
          className={[
            'squarePiece',
            value ? (value === 'Black' ? 'squareBlack' : 'squareWhite') : 'hide'
          ].join(' ')}
        />
      </div>
      {/* <div className={['squareItem', props.value ? (props.value === 'O' ? 'squareItemO' : 'squareItemX') : ''].join(' ')}></div> */}
    </div>
  );
}
