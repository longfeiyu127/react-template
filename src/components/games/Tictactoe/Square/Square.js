import React from 'react';
// import PropTypes from 'prop-types';
import './Square.less';

function TictactoeSquare(props) {
  // console.log(props)
  // eslint-disable-next-line
  const { onClick, heightLine, value } = props;
  console.log(props);
  const squareItem = value
    ? value === 'O'
      ? 'squareItemO'
      : 'squareItemX'
    : '';
  return (
    <div
      className="g-tic-square"
      onClick={onClick}
      style={{ color: heightLine ? 'red' : '' }}
    >
      <div className={['squareItem', squareItem].join(' ')} />
    </div>
  );
}

// TictactoeSquare.propTypes = {
//   onClick: PropTypes.func.isRequired,
//   heightLine: PropTypes.oneOf([PropTypes.bool, null]).isRequired,
//   value: PropTypes.string.isRequired
// };

export default TictactoeSquare;
