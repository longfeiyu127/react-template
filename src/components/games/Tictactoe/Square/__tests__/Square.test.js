import React from 'react';
import renderer from 'react-test-renderer';
import { TictactoeSquare } from '../Square';

describe('<TictactoeSquare />', () => {
  it('component render', () => {
    let victory = renderer
      .create(<TictactoeSquare heightLine value="O" onClick={() => {}} />)
      .toJSON();
    expect(victory).toMatchSnapshot();

    victory = renderer.create(<TictactoeSquare onClick={() => {}} />).toJSON();
    expect(victory).toMatchSnapshot();

    victory = renderer
      .create(<TictactoeSquare value="X" onClick={() => {}} />)
      .toJSON();
    expect(victory).toMatchSnapshot();
  });
});
