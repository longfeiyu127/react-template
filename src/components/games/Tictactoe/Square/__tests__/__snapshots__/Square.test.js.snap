// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TictactoeSquare /> component render 1`] = `
<div
  className="g-tic-square"
  onClick={[Function]}
  style={
    Object {
      "color": "red",
    }
  }
>
  <div
    className="squareItem squareItemO"
  />
</div>
`;

exports[`<TictactoeSquare /> component render 2`] = `
<div
  className="g-tic-square"
  onClick={[Function]}
  style={
    Object {
      "color": "",
    }
  }
>
  <div
    className="squareItem "
  />
</div>
`;

exports[`<TictactoeSquare /> component render 3`] = `
<div
  className="g-tic-square"
  onClick={[Function]}
  style={
    Object {
      "color": "",
    }
  }
>
  <div
    className="squareItem squareItemX"
  />
</div>
`;
